java/integTest/src/test/java/glide/ConnectionTests.java

```diff
@@ -410,4 +426,27 @@ public void test_az_affinity_replicas_and_primary_routes_to_primary() {
 
         azTestClient.close();
     }
+
+    /**
+     * Test that the client can connect using both secure and insecure TLS modes, meaning the client
+     * bypasses the SSL certificate validation.
+     */
+    @SneakyThrows
+    @ParameterizedTest
+    @CsvSource(value = {"true, true", "true, false", "false, true", "false, false"})
+    public void test_connection_tls_mode(String clusterMode, String insecureTls) {
```

Comment:
The cluster started for IT doesn't use TLS.
Ideally we need to do another PR where we move all cluster start/stop logic from `build.gradle` into java code. We will be able to manage clusters from withing tests, so we will be able to run another cluster with TLS (with self-signed certificates) and validate how this feature works.
Before that done this test is no-op.